src/routes/login/+page.svelte

```diff
@@ -59,7 +59,7 @@
       }
       break;
     default:
-      redirectBuilder = () => decodeURIComponent(redirectParam);
+      redirectBuilder = () => `${decodeURIComponent(redirectParam)}${allOtherParams ? `?${allOtherParams}` : ''}`;
```

Comment:
I noticed that we were dropping query params when redirecting within the site. I don't think that was intentional so I added this. Hopefully it doesn't break any login flow.